services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,60 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+  if(event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  } else if(thermalMgrQueueHandle == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  } else {
+    BaseType_t errCode = xQueueSend(thermalMgrQueueHandle, event, 0);
+    if(errCode != pdTRUE) {
+      LOG_ERROR_CODE(errCode);
+    }
+    return ERR_CODE_SUCCESS;
+  }
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OS_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
   /* Implement this task */
   while (1) {
-    
+    thermal_mgr_event_t data = {0};
+
+    BaseType_t errCodeEvent = xQueueReceive(thermalMgrQueueHandle, pvParameters, portMAX_DELAY);
+
+    data = *(thermal_mgr_event_t *) pvParameters;
+
+    if(errCodeEvent == pdTRUE) {
+      float temp;
+      if (data.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+        error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp); 
+        if(errCode != ERR_CODE_SUCCESS) {
+          LOG_ERROR_CODE(errCode);
+          continue;
+        }
+        addTemperatureTelemetry(temp);
+      } else if(data.type == THERMAL_MGR_EVENT_OS_INTERRUPT) {
+        error_code_t errCode = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &temp); 
+        if(errCode != ERR_CODE_SUCCESS) {
+          LOG_ERROR_CODE(errCode);
+          continue;
+        }
+
+        if(temp > LM75BD_DEFAULT_OT_THRESH) {
+          overTemperatureDetected();
+        } else if(temp < LM75BD_DEFAULT_HYST_THRESH) {
+          safeOperatingConditions();
+        }
+      }
+    } else {
+        LOG_ERROR_CODE(errCodeEvent);
```

Comment:
You should be logging an invalid event (you will need to create or search for it in our sys/errors.h) as the type is different